MetaMask/metamask-extension
# Remove 'Verify contract details' link on Sig Req screen when 'verifyingContract' is absent

Fixes: https://github.com/MetaMask/metamask-extension/issues/17116

## Explanation
When `verifyingContract` is not present in `txData.msgParams.data.domain`, we should not show the navigation link to the verifying contract view. 

Though this piece does not address the issue (With the screen no longer reachable the error mentioned in the issue isn't thrown), `ellipsify()` has been hardened against `undefined` input, with tests added for the function.

### Before
![error-page](https://user-images.githubusercontent.com/54408225/211546529-7266e7fe-00a1-4eff-be1d-bcf2870a39c3.png)

https://user-images.githubusercontent.com/54408225/211546491-9f18a147-2719-4c48-a7cc-680543ff0511.mp

### After
<img width="363" alt="Screen Shot 2023-01-10 at 4 55 07 PM" src="https://user-images.githubusercontent.com/8732757/211689368-14628212-72b0-496d-97bf-79a385d7031a.png">



## Manual Testing Steps
1. Clone test dapp https://github.com/MetaMask/test-dapp
2. Remove `verifyingContract` fields from the Sign Typed Data (it does not matter which version)
3. Run the test dapp project `npm start`
4. Open localhost:9011
5. Connect MM to the test dapp
6. Click on Sign for the Typed Data version that you modified
7. Confirm that the `Verify contract details` link is not present
8. Repeat steps with `verifyingContract` details present
9. Ensure that the `Verify contract details` link is present on the sig req confirmation screen, and can be clicked through as expected.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
